src/math/Quaternion.js

```diff
@@ -108,28 +108,50 @@ Quaternion.prototype.toAxisAngle = function(targetAxis){
     return [targetAxis,angle];
 };
 
-var sfv_t1 = new Vec3(),
-    sfv_t2 = new Vec3();
 
 /**
  * Set the quaternion value given two vectors. The resulting rotation will be the needed rotation to rotate u to v.
  * @method setFromVectors
  * @param {Vec3} u
  * @param {Vec3} v
+ * @author Robert Eisele, https://github.com/rawify
  */
 Quaternion.prototype.setFromVectors = function(u,v){
-    if(u.isAntiparallelTo(v)){
-        var t1 = sfv_t1;
-        var t2 = sfv_t2;
 
-        u.tangents(t1,t2);
-        this.setFromAxisAngle(t1,Math.PI);
+    // Implements https://raw.org/proof/quaternion-from-two-vectors/
+
+    var ux = u.x;
+    var uy = u.y;
+    var uz = u.z;
+
+    var vx = v.x;
+    var vy = v.y;
+    var vz = v.z;
+
+    var uLen = u.norm();
+    var vLen = v.norm();
+
+    if (uLen > 0) ux /= uLen, uy /= uLen, uz /= uLen;
+    if (vLen > 0) vx /= vLen, vy /= vLen, vz /= vLen;
+
+    var dot = ux * vx + uy * vy + uz * vz;
+
+    // Parallel check
+    if (dot > 0.999999) {
+      this.x = 0;
+      this.y = 0;
+      this.z = 0;
+      this.w = 1;
+    } else 
+
+    // Close to PI - antiparallel check
+    if (dot < -0.999999) {
+        this.setFromAxisAngle(Math.abs(ux) > Math.abs(uz) ? new Vec3(-uy, ux, 0) : new Vec3(0, -uz, uy), Math.PI);
```

Comment:
Why provoke GC? Previous solution is GC free.

Reply:
I think the code is cleaner this way and it only handles the (hopefully) super rare edge case this way. But I mean, we could also have an initialized Vec3 somewhere floating around for such operations

Reply:
Yes, rare or not, I'm following 100% GC free principle. This allows me to set breakpoints in Vec/Quat constructors to see where allocations are happening under stress testing, so I can make sure everything is as optimized as it gets (memory-wise). But yea, everyone has different ideas (as usual).

Reply:
Yea good way of seeing it for sure. Should I change the commit to be GC free to be merged or don't you see the necessity?